edx/commerce-coordinator
# REV-2701: decouple order history endpoint from calling ecommerce

**Description:** Our existing order history endpoint in commerce-coordinator calls the EcommerceUserOrdersView class directly (assuming all orders come from ecommerce). Let's decouple the endpoint from the orders source to make this more flexible down the road: 
- make a filter with pipeline steps (configured in base.py), to call different sources and combine the results (in the `orders` app to start)
- make a new component in charge of calling ecommerce (called `ecommerce`)
- move the existing EcommerceApiClient and BaseEdxOAuthClient into their suitable homes (`ecommerce` and `core`), and move the ecommerce pipeline steps into `ecommerce` too
- rename `orders` to `frontend-app-ecommerce`

**JIRA:** [REV-2701](https://2u-internal.atlassian.net/browse/REV-2701)
**Data flow diagram:** https://lucid.app/lucidchart/d383a3a7-ba13-40eb-984b-b09899a01779/edit?invitationId=inv_aa2ee211-7ec7-4b9a-8697-0e59bfcfea16&page=tGpbPtsf1hqT#

**Testing instructions:**
Test on commerce-coordinator: 
1. Get commerce-coordinator running locally (as per [README](https://github.com/edx/commerce-coordinator) instructions), and log in as edx user on localhost
2. Go to order status test page for edx user: http://localhost:8140/orders/get_order_data/?username=edx&page=1&page_size=20
Result: orders still come through as before, now using this new pipeline

Test on frontend-app-ecommerce: 
1. set up F-A-ecommerce to go through commerce-coordinator for orders (waffle flag on and base url configured)
2. to go the local orders page for this user: http://localhost:1996/orders
3. Wait for it to load (this takes multiple minutes locally) and view the developer console to see the output that comes back first from ecommerce then from commerce-coordinator
Result: the order history page loads successfully
<img width="1452" alt="order history page via commerce-coordinator" src="https://user-images.githubusercontent.com/5384216/180270811-9bcb4aec-ce6f-4275-aa66-45e5aede9784.png">

**Merge checklist:**
